File: data/src/main/java/org/cryptomator/data/cloud/s3/RootS3Folder.java
package org.cryptomator.data.cloud.s3;

import org.cryptomator.domain.Cloud;
import org.cryptomator.domain.S3Cloud;

class RootS3Folder extends S3Folder {

	private final S3Cloud cloud;

	public RootS3Folder(S3Cloud cloud) {
		super(null, "", "");
		this.cloud = cloud;
	}

	@Override
	public S3Cloud getCloud() {
		return cloud;
	}

	@Override
	public S3Folder withCloud(Cloud cloud) {
		return new RootS3Folder((S3Cloud) cloud);
	}
}

File: data/src/main/java/org/cryptomator/data/cloud/s3/RootS3Folder.kt
package org.cryptomator.data.cloud.s3

import org.cryptomator.domain.Cloud
import org.cryptomator.domain.S3Cloud

internal class RootS3Folder(override val cloud: S3Cloud) : S3Folder(null, "", "") {

	override val key: String
		get() = ""

	override fun withCloud(cloud: Cloud?): S3Folder {
		return RootS3Folder(cloud as S3Cloud)
	}
}
